lib/internal/fs/cp/cp.js

```diff
@@ -63,15 +63,16 @@ async function cpFn(src, dest, opts) {
     process.emitWarning(warning, 'TimestampPrecisionWarning');
   }
   const stats = await checkPaths(src, dest, opts);
-  const { srcStat, destStat } = stats;
+  const { srcStat, destStat, skipped } = stats;
+  if (skipped) return;
   await checkParentPaths(src, srcStat, dest);
-  if (opts.filter) {
-    return handleFilter(checkParentDir, destStat, src, dest, opts);
-  }
   return checkParentDir(destStat, src, dest, opts);
 }
 
 async function checkPaths(src, dest, opts) {
+  if (opts.filter && !(await opts.filter(src, dest))) {
+    return { skipped: true };
```

Comment:
I think the following would crash:

```js
Object.defineProperty(Object.prototype, 'srcStat', { get() { throw new Error } });
fs.cp(src, dest, { filter() { return true } }, () => {});
```

We can avoid that by using a null-prototype object here:

```suggestion
    return { __proto__: null, skipped: true };
```

Reply:
Thanks for pointing this out @aduh95